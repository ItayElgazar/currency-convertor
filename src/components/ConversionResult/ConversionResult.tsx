import React, { FC } from 'react';
import { Typography } from '@material-ui/core';

type ConversionResultProps = {
  fromCurrency: string;
  toCurrency: string;
  amount: number;
  convertedAmount: number;
};

const ConversionResult: FC<ConversionResultProps> = ({
  fromCurrency,
  toCurrency,
  amount,
  convertedAmount
}) => (
  <Typography variant="h4" component="p" align="center">
    {amount} {fromCurrency} ={' '}
    <Typography variant="h4" component="span" color="secondary">
      {convertedAmount}
    </Typography>{' '}
    {toCurrency}
  </Typography>
);

export default ConversionResult;
