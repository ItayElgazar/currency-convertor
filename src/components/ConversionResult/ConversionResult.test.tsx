import React from 'react';
import ConversionResult from './ConversionResult';
import renderer from 'react-test-renderer';

describe('#ConversionResult', () => {
  it('should render correctly', () => {
    const conversionResult = renderer
      .create(
        <ConversionResult
          fromCurrency="EURO"
          toCurrency="ILS"
          amount={1}
          convertedAmount={4}
        />
      )
      .toJSON();
    expect(conversionResult).toMatchSnapshot();
  });
});
