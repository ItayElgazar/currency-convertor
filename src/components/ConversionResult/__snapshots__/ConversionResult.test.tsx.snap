// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#ConversionResult renders correctly 1`] = `
<p
  className="MuiTypography-root MuiTypography-h4 MuiTypography-alignCenter"
>
  1
   
  EURO
   =
   
  <span
    className="MuiTypography-root MuiTypography-h4 MuiTypography-colorSecondary"
  >
    4
  </span>
   
  ILS
</p>
`;

exports[`#ConversionResult should render correctly 1`] = `
<p
  className="MuiTypography-root MuiTypography-h4 MuiTypography-alignCenter"
>
  1
   
  EURO
   =
   
  <span
    className="MuiTypography-root MuiTypography-h4 MuiTypography-colorSecondary"
  >
    4
  </span>
   
  ILS
</p>
`;
